### ISSUE TITLE
v0.10 release

### ISSUE THREAD

**Issue by vsalvino** [coderedcorp/coderedcms#54]
Ok, time to get this out the door.

Once all of the current issues with milestone 0.10 are merged in we will call it quits and not add any more issues to this release.

For this release in particular, there are a number of migrations and migration issues. So I plan to delete all of the migrations created between 0.9.1 and now, and then re-generate one clean migration with all of the changes. That way existing sites should have a smooth upgrade to 0.10.

Finally, with the new documentation, I also plan to document as much of the new changes as possible after they are merged in, before we cut the release.

Any objections or concerns, speak now or forever hold your peace! (until 0.11 at least)

**Comment by FlipperPA** [comment]
There seem to be some CSS issues when integrating with other page types. Here's an example.

Without CodeRedCMS:

![image](https://user-images.githubusercontent.com/68164/50058133-22832480-0142-11e9-9a5b-fec8a9969369.png)

With CodeRedCMS:

![image](https://user-images.githubusercontent.com/68164/50058146-3f1f5c80-0142-11e9-8880-e800fde112da.png)

It looks like everything is embiggened (to use a classic Simpsons term) but some CSS included in the CodeRedCMS payload. Will investigate further, but figured I'd get it reported.

**Comment by FlipperPA** [comment]
When I remove the `codered-admin.css` and `codered-editor.css`, the page looks like it should (of course). I'm curious, what CSS do you need above and beyond the typical Wagtail editor CSS, beyond the powered by CodeRed? This could lead to a lot of maintenance down the road.

**Comment by vsalvino** [comment]
Yes, we do have quite a bit of custom CSS in the wagtail admin. Most of this is to support our deeply nested streamfields, so we actually had to do quite a bit of CSS hacking to make this work. If you load up a content-heavy `CoderedWebPage` in the admin, with and without the codered CSS, you will see that it is impossible to use without the custom CSS.

Some additional "custom" changes are accessibility/readability related: Use system font, use system font size, and reduce contrast with both font and input colors. If you've ever used wagtail on a "standard issue" Windows machine with a bad screen or Windows 7, it actually looks pretty terrible and unreadable. Windows is notorious for bad font rendering at small sizes, and most low quality displays actually make the light-colored inputs completely dissapear. Of course it looks good on a mac or high quality display favored by developers, but editors are usually not so lucky!

Some of these changes should probably be pull-requested into wagtail directly, although it is very opinionated stuff.

So to quote everybody everywhere, "it's not a bug, it's a feature" :)

**Comment by vsalvino** [comment]
That being said, if something actually looks broken on your existing pages, or has bad alignment, please do open an issue for that.

**Comment by FlipperPA** [comment]
I'll open up an issue for the bad alignment above. As for the accessibility / readability concerns, we may be able to remove that CSS, thank to this: https://github.com/wagtail/wagtail/pull/4624 and this: https://github.com/wagtail/wagtail/pull/4788

**Comment by vsalvino** [comment]
Well, our css has been in there since the beginning, so it would require some significant work to remove without breaking our streamfield. 

I have noticed wagtail has been making some improvements to accessibility, unfortunately that does not apply to any form inputs it seems. Luckily the stuff we customized appears mostly unrelated to the changes you cited in wagtail.

The most noticeable difference remains the system font and font size. Personally I don't agree with wagtail admin requiring multiple weights of open sans and robots slab when perfectly good system fonts are available. The admin is heavy enough, why pile fonts on top of that when the fonts will not be matching the brand's front-end anyhow? Another personal pet peeve is when web apps do not respect the browser's font size and insist on shrinking it down. That's all personal opinion though. All of this will have to be reviewed before 1.0, which I appreciate you providing this feedback with an existing project, it is valuable. Do open an issue with highlighted screenshots, it would be appreciated. 

I would not hold up for Wagtail 2.4, that can come in a future release. We tend to do releases every one or two months to keep up with the needs of our client projects. Plus 2.3 and django 1.11 make a nice LTS combination.
